### PR TITLE
Parser clean

### PR DESCRIPTION
**Pull request by Ericson2314**
A few misc shortenings, mainly from my stdlib deps prototype. See the TODOs for some questions.

### CODE REVIEW

File: src/cargo/util/toml.rs [hunk 1]
```diff
@@ -551,26 +550,38 @@ impl TomlManifest {
...
-                                              Some(Kind::Development)));
+
+            fn process_dependencies(
+                cx: &mut Context,
+                ds: [&Option<HashMap<String, TomlDependency>>; 3])
+                -> CargoResult<()>
+            {
+                let dep_iter = ds[0].iter().map(|d| (d, None))
+                    .chain(ds[1].iter().map(|d| (d, Some(Kind::Development))))
+                    .chain(ds[2].iter().map(|d| (d, Some(Kind::Build))))
```
Comment (alexcrichton) [inline src/cargo/util/toml.rs]: This seems a little magical to me, could the previous behavior of passing this in explicitly be kept instead?

File: src/cargo/util/toml.rs [hunk 2]
```diff
@@ -765,42 +770,38 @@ impl TomlDependency {
...
-                        if cx.source_id.is_path() {
-                            let path = cx.layout.root.join(path);
-                            let path = util::normalize_path(&path);
-                            Some(try!(SourceId::for_path(&path)))
-                        } else {
-                            Some(cx.source_id.clone())
-                        }
-                    }
-                    None => None,
+        // TODO(@Ericson2314): Warn if both git and path is specified?
```
Comment (alexcrichton) [inline src/cargo/util/toml.rs]: Can you remove these TODO annotations and open issues instead?

**Comment by Ericson2314** [comment]
Both fixed! Issue is https://github.com/rust-lang/cargo/issues/2869

**Comment by alexcrichton** [comment]
@bors: r+ b4379aa069b6090fa458873658a5b885a8f1304a

Thanks!

**Pull request merged**